lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    return `//${hostname}${pathname}`;
+  } else {
+    // Otherwise, it's a local path that requires a drive letter
+    var letter = pathname.codePointAt(1) | 0x20;
+    var sep = pathname[2];
+    if (letter < 97 || letter > 122 ||   // a..z A..Z
+        (sep !== ':' && sep !== '|')) {
+      return new Error('File URLs must specify absolute paths');
+    }
+    pathname = pathname.slice(1);
+    for (var n = 0; n < pathname.length; n++) {
+      if (pathname[n] === '%') {
+        var third = pathname.codePointAt(n + 2) | 0x20;
+        if ((pathname[n + 1] === '2' && third === 102) || // 2f 2F /
+            (pathname[n + 1] === '5' && third === 99)) {  // 5c 5C \
+          return new Error('Path must not include encoded \\ or / characters');
+        }
+      }
+    }
+    return decodeURIComponent(pathname);
```

Comment:
We cannot. Consider the following URL: `file:///c:/%2f%5c/foo`. Here the `pathname` component is `/c:/%2f%5c/`, if we decode the `pathname` first, it would become `/c://\/`, making it impossible to distinguish the decoded `/` from the legitimate `/` path separators that already exist. The only approach that works is to scan for the %2f and %5c patterns first and *then* decode.